Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/MenuViewController.swift

```diff
@@ -1,405 +1,287 @@
+    init<MenuContent: View>(@ViewBuilder content: (MenuViewController) -> MenuContent) {
+        super.init(nibName: nil, bundle: nil)
+        viewControllers = [MenuHostingController(content: content(self))]
+        menuNavigationDelegate = MenuNavigationControllerDelegate(panModal: self)
+        delegate = menuNavigationDelegate
     }
     
     @available(*, unavailable)
     required init(coder: NSCoder) {
         fatalError()
     }
-    @available(*, unavailable)
-    override var textLabel: UILabel? {
-        return nil
-    }
-    @available(*, unavailable)
-    override var imageView: UIImageView? {
-        return nil
-    }
-}
-
-class MenuViewController: UITableViewController {
     
-    private struct UX {
-        static let rowHeight: CGFloat = 45
-        static let separatorColor = UIColor(white: 0.0, alpha: 0.1)
-        static let topBottomInset: CGFloat = 5
-    }
-    
-    private enum MenuButtons: Int, CaseIterable {
-        case vpn, settings, history, bookmarks, downloads, add, share
-        
-        var title: String {
-            switch self {
-            // This string should not be translated.
-            case .vpn: return "Brave VPN"
-            case .bookmarks: return Strings.bookmarksMenuItem
-            case .history: return Strings.historyMenuItem
-            case .settings: return Strings.settingsMenuItem
-            case .add: return Strings.addToMenuItem
-            case .share: return Strings.shareWithMenuItem
-            case .downloads: return Strings.downloadsMenuItem
+    func presentInnerMenu(_ viewController: UIViewController,
+                          expandToLongForm: Bool = true) {
+        let container = InnerMenuNavigationController(rootViewController: viewController)
+        container.delegate = menuNavigationDelegate
+        container.modalPresentationStyle = .overCurrentContext // over to fix the dismiss animation
+        container.innerMenuDismissed = {
+            if !self.isDismissing {
+                self.panModalSetNeedsLayoutUpdate()
             }
         }
-        
-        var icon: UIImage {
-            switch self {
-            case .vpn: return #imageLiteral(resourceName: "vpn_menu_icon").template
-            case .bookmarks: return #imageLiteral(resourceName: "menu_bookmarks").template
-            case .history: return #imageLiteral(resourceName: "menu-history").template
-            case .settings: return #imageLiteral(resourceName: "menu-settings").template
-            case .add: return #imageLiteral(resourceName: "menu-add-bookmark").template
-            case .share: return #imageLiteral(resourceName: "nav-share").template
-            case .downloads: return #imageLiteral(resourceName: "menu-downloads").template
+        present(container, animated: true) {
+            self.panModalSetNeedsLayoutUpdate()
+        }
+        if expandToLongForm {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.15) {
```

Comment:
why do we delay while xpanding to long form ?

Reply:
Just makes the animation look nicer, I'll make a comment explaining 🙂